java-frontend/src/main/java/org/sonar/java/se/ProgramState.java

```diff
@@ -50,21 +50,64 @@
 import java.util.Objects;
 import java.util.Set;
 import java.util.function.BiConsumer;
+import java.util.stream.Collectors;
 
 public class ProgramState {
 
   public static class Pop {
 
     public final ProgramState state;
     public final List<SymbolicValue> values;
+    public final List<SymbolicValueSymbol> valuesAndSymbols;
 
-    public Pop(ProgramState programState, List<SymbolicValue> result) {
+    public Pop(ProgramState programState, List<SymbolicValueSymbol> result) {
       state = programState;
-      values = result;
+      values = result.stream().map(SymbolicValueSymbol::symbolicValue).collect(Collectors.toList());
+      valuesAndSymbols = result;
     }
 
   }
 
+  public static class SymbolicValueSymbol {
+    final SymbolicValue sv;
+    final Symbol symbol;
+
+    public SymbolicValueSymbol(SymbolicValue sv, @Nullable Symbol symbol) {
+      this.sv = sv;
+      this.symbol = symbol;
+    }
+
+    public SymbolicValue symbolicValue() {
+      return sv;
+    }
+
+    public Symbol symbol() {
+      return symbol;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+      if (this == o) {
+        return true;
+      }
+      if (o == null || getClass() != o.getClass()) {
+        return false;
+      }
+      SymbolicValueSymbol that = (SymbolicValueSymbol) o;
+      return Objects.equals(sv, that.sv);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(sv);
+    }
+
+    @Override
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS864)